### PR TITLE
chore(testing): add ability to skip GAE Flex tests

### PR DESCRIPTION
**Pull request by ace-n**
Related to b/173545239 and #1242

### CODE REVIEW

File: .kokoro/common.cfg [hunk 1]
```diff
@@ -14,3 +14,9 @@ env_vars: {
     key: "TRAMPOLINE_BUILD_FILE"
     value: "github/php-docs-samples/.kokoro/system_tests.sh"
 }
+
+# Skip GAE Flexible deployment tests.
+env_vars: {
+    key: "RUN_APPENGINE_FLEX_TESTS"
+    value: "false"
+}
```
Comment (bshaffer) [inline .kokoro/common.cfg:22]: We can remove this one!

File: appengine/flexible/datastore/test.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export RUN_APPENGINE_FLEX_TESTS=false
+export DIR=$(pwd)
+
+if [ "$RUN_APPENGINE_FLEX_TESTS" != "true" ] &&
+       [[ -n $(grep -ri 'env:.+flex' app.*ml $DIR ) ]]; then
+        echo "Skipping tests in $DIR (no App Engine Flex tests)"
+        continue
+fi
```
Comment (bshaffer) [inline appengine/flexible/datastore/test.sh:8]: Do we need this file? I don't think so.

**Review by bshaffer — CHANGES_REQUESTED** [review]
Hmm, so we don't actually want to skip *all* flex tests. We just want to skip flex DEPLOYMENT tests. I think we could do this by:

1. Adding a check in each GAE flex deployment test to verify `RUN_FLEX_DEPLOYMENT_TESTS` should be `true`. This could be done by adding this to every `appengine/flexible/*/test/DeployTest.php` file:
    ```php
    /** @beforeClass */
    public static function checkProjectEnvVarBeforeClass()
    {
        self::requireEnv('RUN_FLEX_DEPLOYMENT_TESTS');
        self::checkProjectEnvVars();
    }
    ```

2. Add a block to `run_test_suite.sh` to temporarily set `RUN_DEPLOYMENT_TESTS` to false if the directory contains an `app.yaml` with `env: flex`. I can't really see a good/elegant way to do that with the current script though, so I prefer option 1

**Comment by ace-n** [comment]
> Add a block to run_test_suite.sh to temporarily set RUN_DEPLOYMENT_TESTS to false if the directory contains an app.yaml with env: flex. I can't really see a good/elegant way to do that with the current script though, so I prefer option 1

We already do this - that's what the `grep` in `run_test_suite.sh` is for. 🙂

File: appengine/flexible/datastore/test.sh (hunk 2 above)
Comment (ace-n) [reply appengine/flexible/datastore/test.sh]: Good catch - reverted.

File: .kokoro/common.cfg (hunk 1 above)
Comment (ace-n) [reply .kokoro/common.cfg]: Done.

**Review by bshaffer — APPROVED** [review]
Looks great! Should we skip the deployment test in endpoints/getting-started as well? I believe so!

**Pull request merged**